pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -336,3 +345,51 @@ func (hdl openstackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	w.WriteHeader(http.StatusNotFound)
 }
+
+// ServeHTTP for kubeConfigHandler provides cluster kube config
+func (hdl kubeConfigHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		msg := "kube config handler: request method is not Post"
+		log.Error(msg)
+		http.Error(w, msg, http.StatusMethodNotAllowed)
+		return
+	}
+	if r.Header.Get("Content-Type") != "application/json" {
+		msg := "kube config handler: Content-Type header is not application/json"
+		log.Error(msg)
+		http.Error(w, msg, http.StatusUnsupportedMediaType)
+		return
+	}
+	// Use http.MaxBytesReader to enforce a maximum read of 32KB from the
+	// response body. A request body larger than that will now result in
+	// Decode() returning a "http: request body too large" error.
+	r.Body = http.MaxBytesReader(w, r.Body, 32768)
+	dec := json.NewDecoder(r.Body)
+	var kubeConfig types.KubeConfig
+	err := dec.Decode(&kubeConfig)
+	if err != nil {
+		switch {
+		case err.Error() == "http: request body too large":
+			msg := "kube config handler: request body must not be larger than 1MB"
```

Comment:
I also made a comment/question a week and half ago asking whether the kube data is already json or whether this should support arbitrary bytes, plus asking for a more explicit check on the size limit.
My guess is that this implicitly verifies the size limit due to a truncated message resulting in invalid json, but I think we want an API which is just bytes, and an explicit size check hence no json decode here. But should we discuss in case I'm missing something.

(And I have no idea how github lost those two comments! Sigh)

Reply:
Turns out those comments were in slack and not github.